src/prefect/task_server.py

```diff
@@ -144,43 +150,52 @@ async def _submit_pending_task_run(self, task_run: TaskRun):
             f"Submitting run {task_run.name!r} of task {task.name!r} to engine"
         )
 
-        task_run.tags = distinct(task_run.tags + list(self.tags))
+        task_run.tags = distinct(task_run.tags + list(self.extra_tags))
 
         self._runs_task_group.start_soon(
             partial(
                 submit_autonomous_task_to_engine,
                 task=task,
                 task_run=task_run,
                 parameters=parameters,
+                task_runner=self.task_runner,
+                client=self._client,
             )
         )
 
     async def __aenter__(self):
         logger.debug("Starting task server...")
-        self._client = get_client()
-        await self._client.__aenter__()
-        await self._runs_task_group.__aenter__()
+
+        self._client = await self._exit_stack.enter_async_context(get_client())
+        await self._exit_stack.enter_async_context(self._runs_task_group)
+        await self._exit_stack.enter_async_context(self.task_runner.start())
 
         self.started = True
         return self
 
     async def __aexit__(self, *exc_info):
         logger.debug("Stopping task server...")
         self.started = False
-        if self._runs_task_group:
-            await self._runs_task_group.__aexit__(*exc_info)
-        if self._client:
-            await self._client.__aexit__(*exc_info)
+
+        await self._exit_stack.__aexit__(*exc_info)
 
 
 @sync_compatible
-async def serve(*tasks: Task, tags: Optional[Iterable[str]] = None):
-    """Serve the provided tasks so that they may be executed autonomously.
+async def serve(
+    *tasks: Task,
+    task_runner: Optional[Type[BaseTaskRunner]] = None,
+    extra_tags: Optional[Iterable[str]] = None,
+):
+    """Serve the provided tasks so that they may be submitted and executed to the engine.
+    Tasks do not need to be within a flow run context to be submitted and executed.
+    Ideally, you should `.submit` the same task object that you pass to `serve`.
 
     Args:
         - tasks: A list of tasks to serve. When a scheduled task run is found for a
             given task, the task run will be submitted to the engine for execution.
-        - tags: A list of tags to apply to the task server. Defaults to `["autonomous"]`.
+        - task_runner: The task runner to use for executing the tasks. Defaults to
+            `ConcurrentTaskRunner`.
+        - tags: A list of tags to add to task runs submitted by the task server.
```

Comment:
Are we still doing the tags thing?  The TaskServer doesn't submit tasks anymore, right?

Reply:
i was thinking I'd keep the tags there for now, just thinking forward to:
- related resources on events
- rendering them in the UI
- concurrency limits

but they are not needed as far as identifying task runs

Reply:
rm'd in [9eab2be](https://github.com/PrefectHQ/prefect/pull/11806/commits/9eab2be0b1d14f3472269923f9ab67aba45ae322)